### PR TITLE
snippet parser

### PR DESCRIPTION
**Pull request by mjauvin**
Add missing code to parse markup for snippets and render the snippets.

### CODE REVIEW

**Comment by bennothommo** [comment]
@mjauvin is this still in progress? The title of the PR still contain "wip".

**Comment by mjauvin** [comment]
> @mjauvin is this still in progress? The title of the PR still contain "wip".

Yes, it's ready!

File: classes/SnippetLoader.php [hunk 1]
```diff
@@ -0,0 +1,181 @@
+<?php namespace StudioAzura\W2C\Classes;
```
Comment (bennothommo) [inline classes/SnippetLoader.php:1]: Incorrect namespace here :)
```suggestion
<?php namespace Winter\Pages\Classes;
```

File: classes/SnippetLoader.php [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    public static function registerComponentSnippet($snippetInfo)
+    {
+        $controller = CmsController::getController();
+
+        // Make an unique alias for this snippet based on its name and parameters
+        #$snippetInfo['code'] = uniqid($snippetInfo['code'] . '_' . md5(serialize($snippetInfo['properties'])) . '_');
+        // the line above was commented out to allow the overriden partials in theme to be used for the component alias
+
+        self::attachComponentSnippetToController($snippetInfo, $controller, true);
+        self::cacheSnippet($snippetInfo['code'], $snippetInfo);
```
Comment (bennothommo) [inline classes/SnippetLoader.php:37]: We prefer `static` over `self`, as `static` refers to the extended class if this class is extended upon. Could you also change this for all other `self` references in this file.

```suggestion
        static::attachComponentSnippetToController($snippetInfo, $controller, true);
        static::cacheSnippet($snippetInfo['code'], $snippetInfo);
```

File: classes/SnippetLoader.php [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    /**
+     * Get cached component snippets from the cache.
+     *
+     * @param CmsPage $page         The CMS page for which to load the cache
+     */
+    protected static function fetchCachedSnippets(CmsPage $page)
+    {
+        $cache = @unserialize(Cache::get(self::getMapCacheKey($page), serialize([])));
```
Comment (bennothommo) [inline classes/SnippetLoader.php:164]: Any reason why the error is squelched here? (the `@` symbol)

**Review by bennothommo — CHANGES_REQUESTED** [review]

File: classes/SnippetLoader.php (hunk 3 above)
Comment (mjauvin) [reply classes/SnippetLoader.php:164]: @bennothommo Not sure, I only ported this code from another plugin.

**Comment by LukeTowers** [comment]
@bennothommo this is on my list to review, feel free to review it as well but don't merge yet please :)

File: classes/SnippetLoader.php (hunk 3 above)
Comment (LukeTowers) [reply classes/SnippetLoader.php]: @bennothommo any other changes you want here?

Comment (bennothommo) [reply classes/SnippetLoader.php]: Yes, I would prefer the error squelching be removed so that the error can be seen if it occurs.

Comment (mjauvin) [reply classes/SnippetLoader.php]: Better like this @bennothommo ?

**Review by bennothommo — APPROVED** [review]

**Review by LukeTowers — APPROVED** [review]

**Pull request merged**